### ISSUE TITLE
Compatibility with Rancher Server

### ISSUE THREAD

**Issue by jmatsushita** [scaleway/docker-machine-driver-scaleway#21]
I've submitted an issue on the rancher issue track as the release doesn't install: rancher/rancher#4486

Not sure if this driver has something to do with it. I was able to install the Vultr one.

**Comment by moul** [comment]
Thank you for reporting those issues,

We will wait a little bit an answer from the @rancher-team on their issue

**Comment by jmatsushita** [comment]
Taring with only the go static binary at the root (no folder) did the trick!

**Comment by moul** [comment]
Ok cool, I just added another upload here: https://github.com/scaleway/docker-machine-driver-scaleway/releases/download/v1.0.2/docker-machine-driver-scaleway.tar.gz, you should be able to use the url directly in rancher

**Comment by moul** [comment]
Oups, I gave you the bad link in the other issue, can you try with this link : https://github.com/scaleway/docker-machine-driver-scaleway/releases/download/v1.0.2/docker-machine-driver-scaleway.tar.gz (a tarball with only the go binary without folder)

**Comment by jalberto** [comment]
@moul that url is not working neither 1.2 relase

**Comment by moul** [comment]
@jalberto, yep I only succeed by putting the binary without tarball, we will probably release the binary for linux x86_64 then

**Comment by jalberto** [comment]
It works for me wit a zip file containing only the binary

**Comment by shouze** [comment]
@moul @QuentinPerez @jalberto  ok same issue for me, it only works within a zip (nor tgz) file... :(

Rancher also released as skel project for custom machine drivers:
https://github.com/rancher/ui-driver-skel

Would be cool to provide some skel for scaleway ;)

**Comment by shouze** [comment]
@jalberto don't work when zipping it. Which version of rancher do you use? (I'm using 1.0.1 ATM).

**Comment by jalberto** [comment]
It did work for me on 1.1.0 inside the zip file you only need the binary

Anyway this driver is yet quite limited, I found faster just to use scaleway UI

**Comment by shouze** [comment]
@jalberto ok thx I will try with 1.1.0 ;)

I already use the driver with docker-machine & it fits all my needs. I will just provide a skel following https://github.com/rancher/ui-driver-skel spec.

**Comment by jalberto** [comment]
great :)

**Comment by pascalandy** [comment]
I'm super pumped about Rancher too. I tried it but it's not supporting compose v2 at the moment. I decided to wait and go with Docker stuff only, compose v2, machine & swarm. 

Anyways, cool hear about another folks using scaleway with the docker eco-system!
![beer-alcohol-toast-3o85xjSETVG3OpPyx2](https://media4.giphy.com/media/3o85xjSETVG3OpPyx2/giphy.gif)

Cheers!

**Comment by shouze** [comment]
@pascalandy yes, I will stuck with docker stuff only till rancher-compose support docker-compose v2 format https://github.com/rancher/rancher/issues/3973 (because libcompose itself still not ready for v2 format https://github.com/docker/libcompose/issues/147).

**Comment by clescot** [comment]
Hi,
FYI, [rancher/rancher#3973](https://github.com/rancher/rancher/issues/3973) (in 1.2 pre-2 release) and [docker/libcompose#147](https://github.com/docker/libcompose/issues/147) (in libcompose 0.3.0 ) issues have been fixed.
Charles.

**Comment by shouze** [comment]
@clescot yes that's great but volumes & networks have some issues opened https://github.com/rancher/rancher/issues/5714 & https://github.com/rancher/rancher/issues/5715. BTW, close to the goal ;)

**Comment by RVN-BR** [comment]
Hi, just wanted to know if this is working? and if so what URL are you using in rancher to download the machine driver?

thanks!

**Comment by robeastham** [comment]
@RVN-BR If I'm following you correctly you have managed to get Rancher OS v0.6.1 running on a Scaleway instance? That's great if so. I've been trying to build a custom Rancher OS image for use on Scaleway VC1 or C2's with this:

I already have Rancher hosts running off Ubuntu derived images on Scaleway, but I'm keen to use Rancher OS as the base image for my hosts instead.  I'm hoping that it'll use a lot less memory than the Scaleway default Docker image.

I did try this:

https://github.com/rancher/os-installer

But I've had no luck and just get a kernel panic on boot, think this is because Scaleway ignores vmlinuz and initrd and uses it's own instead. 

If you have managed to get ROS work on Scaleway would you be willing to share the steps required and perhaps and example of your docker-machine line? Thanks :-)

**Comment by RVN-BR** [comment]
Hi @robeastham in fact I didnt get that far :(

I actually wasnt even able to get the docker-machine driver working...  (although I think its a question of looking at the scaleway API and filling in the options correctly).  I'll probably look into this a bit later as I havent had a lot of time...

I was going to have it start an ubuntu image, at least initially... hadnt thought of giving rancher os another try...  I had some troubles with it and then ended up not using it anymore...  Have you been using ROS in production?  thanks!

**Comment by robeastham** [comment]
@RVN-BR no not using in production yet. But hope to soon with my setup. 

I  have a Rancher Server running on an EC2 instance, that is running Rancher OS. But I plan to use Scaleway as much as possible for hosts as they are much better value when compared to EC2 instances.

I've just settled with the default Ubuntu based Docker image for now. Hosts just work with my Rancher Server when I pick this image. I tried the Scaleway Alpine images too, but had problems with K8S containers communicating and starting despite picking the Docker boot scripts.

Would be great to get Rancher OS as a base image supported by Scaleway as I think it will cut down on the memory requirements for the hosts. Perhaps @moul could comment on whether we are likely to see this? I think Scaleway would sell more instance hours if they supported Rancher OS out of the gate. It wouldn't take long for word to get around the Rancher community.

**Comment by RVN-BR** [comment]
Got it @robeastham ...  I am having a bit of trouble with getting hosts to connect to a rancher server on a scaleway instance... I need to create a different set of security group rules I think....  I tried using the internal scaleway IP (rancher agent and rancher server on scaleway) but it didnt work... I keep getting http://10.x.x.x:8080/v1 is inaccessible errors on the rancher log...  (ping works, etc), anyways, thats a diferent issue...  I wanted to use Rancher on scaleway mostly for test loads, unfortunately most of our traffic is in South America and the lag to Scaleway is significant...  But for testing loads I find the scaleway pricing pretty fantastic 👍 

I'm going to give RancherOS another spin soon...  I havent really touched it in a while...  I think a rrancheros image would indeed help a lot for european customers at least...  And the Rancher Community seems pretty active so I guess unless its something very difficult it would be wise for them...  CoreOS is also notably not on their list...  Previously I was tesing mostly on coreos...will have to give ros another shot.

**Comment by meteormanaged** [comment]
Two questions here:

1. Can someone provide a sample config used within Rancher for provisioning a host? I'm getting failures with a plain old exit code 1.

2. Can this repo be updated to add the custom UI options that rancher supports? Might make the provisioning go a little smoother/require less guess work for what's truly needed or not.

http://docs.rancher.com/rancher/v1.2/en/configuration/machine-drivers/

https://github.com/rancher/ui-driver-skel

**Comment by marcbachmann** [comment]
> Can someone provide a sample config used within Rancher for provisioning a host? I'm getting failures with a plain old exit code 1.

For rancher 1.2 I've used
`https://github.com/scaleway/docker-machine-driver-scaleway/releases/download/v1.3/docker-machine-driver-scaleway_1.3_linux_amd64.tar.gz`

There's no ui for that.

For rancher 1.6 there's also a working ui:
download url:
`https://github.com/scaleway/docker-machine-driver-scaleway/releases/download/v1.3/docker-machine-driver-scaleway_1.3_linux_amd64.tar.gz`

ui url:
`https://mitcdh.github.io/ui-driver-scaleway/dist/component.js`

Here's the full configuration:
![image](https://cloud.githubusercontent.com/assets/431376/25781774/2b4cc220-333e-11e7-9758-843466814a91.png)

**Comment by davidoster** [comment]
Well just a follow up not in terms of RancherOS but general use of Rancher.
It appears that the Scaleway machine driver isn't available anymore at least on version 1.6.2
But to be honest this is not a problem!

I am running a simple Rancher cluster as we speak out of VC1S machines with docker instances.
I tried to install on an Ubuntu Xenial image the RancherOS but couldn't make it!
So the master runs of a VC1S image.

Then I added via a custom machine driver a node and it is working as a charm, even though that the Rancher UI says incompatible!!!
http://prntscr.com/fpvikp

**Comment by davidoster** [comment]
Continuing previous comments,
Got it working by simply installing on a VC1S the docker image from imagehub and then after I ssh'ed to the server I executed curl https://releases.rancher.com/install-docker/1.12.sh | sh.
That last command update docker to 1.12.6!
Happy hunting!
